scripts/deploy_assisted_service.sh

```diff
@@ -112,13 +115,29 @@ EOF
     echo "Installation of Assisted Install operator passed successfully!"
 else
     print_log "Updating assisted_service params"
+
+    if [ "${DEBUG_SERVICE}" == "true" ]; then
+        print_log "******* Patching assisted service image with a debuggable code ***** "
+        (cd assisted-service/ && skipper --env-file ../skipper.env make update-local-image -e LOCAL_ASSISTED_ORG=$(minikube ip):5000 -e CONTAINER_BUILD_EXTRA_PARAMS="--cgroup-manager=cgroupfs --storage-driver=vfs --events-backend=file")
+    fi
+
     skipper run discovery-infra/update_assisted_service_cm.py
     (cd assisted-service/ && skipper --env-file ../skipper.env run "make deploy-all" ${SKIPPER_PARAMS} $ENABLE_KUBE_API_CMD $OPENSHIFT_VERSIONS_CMD DEPLOY_TAG=${DEPLOY_TAG} DEPLOY_MANIFEST_PATH=${DEPLOY_MANIFEST_PATH} DEPLOY_MANIFEST_TAG=${DEPLOY_MANIFEST_TAG} NAMESPACE=${NAMESPACE} AUTH_TYPE=${AUTH_TYPE})
 
     add_firewalld_port $SERVICE_PORT
 
     print_log "Starting port forwarding for deployment/${SERVICE_NAME} on port $SERVICE_PORT"
-    wait_for_url_and_run ${SERVICE_BASE_URL} "spawn_port_forwarding_command $SERVICE_NAME $SERVICE_PORT $NAMESPACE $NAMESPACE_INDEX $KUBECONFIG minikube"
+    wait_for_url_and_run ${SERVICE_BASE_URL} "spawn_port_forwarding_command $SERVICE_NAME $SERVICE_PORT $NAMESPACE $NAMESPACE_INDEX $KUBECONFIG minikube undeclared $SERVICE_INTERNAL_PORT"
```

Comment:
this is changing the behavior without running DEBUG. you meant for it to happen?

Reply:
This function redirects the service node port to the host port.
For example: The assisted-service service(k8s service) serves on 8090 but its node port is different and determine in runtime. 
To access the service from outside we should forward the minikube service node port to the host(beaker machine) port.
If debug is enabled the k8s has two ports, 8090 for the service itself and 40000 for the debug.
The old behavior assumed that the service has only one port but it's not true any more.
I changed the spawn_port_forwarding_command function to resolve the node port by the service port.
Here i just added the service port to the call.